src/Cake.Core/IO/IDirectory.cs

```diff
@@ -32,7 +32,8 @@ public interface IDirectory : IFileSystemInfo
         /// Deletes the directory.
         /// </summary>
         /// <param name="recursive">Will perform a recursive delete if set to <c>true</c>.</param>
-        void Delete(bool recursive);
+        /// <param name="force">Will delete read-only files if set to <c>true</c></param>
+        void Delete(bool recursive, bool force);
```

Comment:
@mholo65 don't want to litter code base with allot of interfaces either, so one option could be to get this in a milestone/release whilst we have other breaking changes that need a minimum cake version set, i.e. the bigger roslyn refactoring 0.21:isch time frame.